lib/models/Action.dart

```diff
@@ -209,107 +215,103 @@ CampaignActionType campaignActionTypeFromString(String? s) {
   return t;
 }
 
-Tuple3<String?, String?, String?> generateCampaingActionDesc(
+Tuple3<String?, String?, String?> generateCampaignActionDesc(
     CampaignActionType? t) {
   if (campaignActionTypeData.containsKey(t)) {
     return Tuple3(
         campaignActionTypeData[t]['verb'],
         campaignActionTypeData[t]['pastVerb'],
         campaignActionTypeData[t]['displayName']);
   }
-  return Tuple3("Complete", "Completed", "special action");
+  return const Tuple3("Complete", "Completed", "special action");
 }
 
 class ListCauseAction extends Explorable {
-  int _id;
-  int get id => _id;
-  
-  String _title;
-  String get title => _title;
+  final int id;
+
+  final String title;
 
   /// The type of the action
-  CampaignActionType _type;
-  /// The type of the action 
-  CampaignActionType get type => _type;
+  final CampaignActionType type;
+
   /// The super type is a bigger category than the type - This is used for the
   /// styling.
   ActionType get superType => campaignActionTypeData[type]['type'];
+
   Color get primaryColor => superType.primaryColor;
+
   Color get secondaryColor => superType.secondaryColor;
+
+  Color get tertiaryColor => superType.tertiaryColor;
+
   IconData get icon => superType.icon;
 
   /// The causes that this action is part of
-  List<ListCause> _causes;
+  final List<ListCause> _causes;
+
   // Although at the api level an action can be in many causes, for now we are
   // only showing a single cause in the UI.
-  ListCause? get cause => _causes.length > 0 ? _causes[0] : null;
-  
-  double _time;
-  double get time => _time;
-  String get timeText => timeBrackets.firstWhere((b) => b['maxTime'] > time)['text'];
-
-  DateTime _createdAt;
-  DateTime? _releasedAt;
-  /// Return when this action was released
-  DateTime get releaseTime => _releasedAt == null ? _createdAt : _releasedAt!;
-  bool get isNew => DateTime.now().difference(releaseTime).compareTo(Duration(days: 2)) < 0;
+  // TODO initialize in constructor initializer list if we're sticking to this logic
+  ListCause get cause => _causes[0];
```

Comment:
@JElgar I've made `cause` always return the first element of `_cause`. If we're sticking to this logic then maybe we should initialise `cause` in the constructor directly and remove `_cause` entirely.

Reply:
Yeah, I think that makes sense! Lets do that

Reply:
@JElgar done!